app/migrations/Version20171003103916.php

```diff
@@ -37,7 +37,7 @@ public function up(Schema $schema): void
         $defaultLocale = $this->container->getParameter('locale');
         $productAttributeRepository = $this->container->get('sylius.repository.product_attribute');
 
-        $productAttributes = $productAttributeRepository->findBy(['type' => SelectAttributeType::TYPE]);
+        $productAttributes = $productAttributeRepository->findByTypeForMigration(SelectAttributeType::TYPE);
```

Comment:
Despite the typehint, from the repository, you can fetch query builder. Therefore, I would say it would be better to just write custom method only in this migration file rather than place it in generally available place

Reply:
Yep. Much easier, you perfectly right. I've done the change.